src/test/unit/types/BasicTypeTest.cpp

```diff
@@ -1,108 +1,579 @@
+
+  BasicType::convertArray(ca.dest, NTA_BasicType_UInt64, ca.bufBool,
+                          NTA_BasicType_Bool, 8);
+  ASSERT_TRUE(ca.checkArrayBool<UInt64>(ca.dest)) << "bool to UInt64 conversion";
+
+  //////////////////////// To Real32 ///////////////////////////////////////////////////////////
+
+  BasicType::convertArray(ca.dest, NTA_BasicType_Real32, ca.bufByte,
+                          NTA_BasicType_Byte, 8);
+  ASSERT_TRUE(ca.checkArray<Real32>(ca.dest)) << "Byte to Real32 conversion";
+  // Note: Bytes are char, not unsigned char.
+
+  BasicType::convertArray(ca.dest, NTA_BasicType_Real32, ca.bufInt16,
+                          NTA_BasicType_Int16, 8);
+  ASSERT_TRUE(ca.checkArray<Real32>(ca.dest)) << "Int16 to Real32 conversion";
+
+  BasicType::convertArray(ca.dest, NTA_BasicType_Real32, ca.bufUInt16,
+                          NTA_BasicType_UInt16, 8);
+  ASSERT_TRUE(ca.checkArray<Real32>(ca.dest)) << "UInt16 to Real32 conversion";
+
+  BasicType::convertArray(ca.dest, NTA_BasicType_Real32, ca.bufInt32,
+                          NTA_BasicType_Int32, 8);
+  ASSERT_TRUE(ca.checkArray<Real32>(ca.dest)) << "Int32 to Real32 conversion";
+
+  BasicType::convertArray(ca.dest, NTA_BasicType_Real32, ca.bufUInt32,
+                          NTA_BasicType_UInt32, 8);
+  ASSERT_TRUE(ca.checkArray<Real32>(ca.dest)) << "UInt32 to Real32 conversion";
+
+  BasicType::convertArray(ca.dest, NTA_BasicType_Real32, ca.bufInt64,
+                          NTA_BasicType_Int64, 6);
+  ASSERT_TRUE(ca.checkArray<Real32>(ca.dest)) << "Int64 to Real32 conversion";
+
+  BasicType::convertArray(ca.dest, NTA_BasicType_Real32, ca.bufUInt64,
+                          NTA_BasicType_UInt64, 6);
+  ASSERT_TRUE(ca.checkArray<Real32>(ca.dest)) << "UInt64 to Real32 conversion";
+
+  BasicType::convertArray(ca.dest, NTA_BasicType_Real32, ca.bufReal32,
+                          NTA_BasicType_Real32, 8);
+  ASSERT_TRUE(ca.checkArray<Real32>(ca.dest)) << "Real32 to Real32 conversion";
+
+  BasicType::convertArray(ca.dest, NTA_BasicType_Real32, ca.bufReal64,
+                          NTA_BasicType_Real64, 6);
+  ASSERT_TRUE(ca.checkArray<Real32>(ca.dest)) << "Real64 to Real32 conversion";
+
+  BasicType::convertArray(ca.dest, NTA_BasicType_Real32, ca.bufBool,
+                          NTA_BasicType_Bool, 8);
+  ASSERT_TRUE(ca.checkArrayBool<Real32>(ca.dest)) << "bool to Real32 conversion";
+
+  //////////////////////// To Real64 //////////////////////////////////////////////////////////////
+
+  BasicType::convertArray(ca.dest, NTA_BasicType_Real64, ca.bufByte,
+                          NTA_BasicType_Byte, 8);
+  ASSERT_TRUE(ca.checkArray<Real64>(ca.dest)) << "Byte to Real64 conversion";
+  // Note: Bytes are char, not unsigned char.
+
+  BasicType::convertArray(ca.dest, NTA_BasicType_Real64, ca.bufInt16,
+                          NTA_BasicType_Int16, 8);
+  ASSERT_TRUE(ca.checkArray<Real64>(ca.dest)) << "Int16 to Real64 conversion";

```

Comment:
Very nice to have these tests! 

I am missing one that tests the error behavior? ei converting <int>{-1,0,1,2} to UInt.